packages/native/types/base.d.ts

```diff
@@ -64,7 +64,7 @@ export interface FilteringStyledOptions<
   shouldForwardProp?(propName: PropertyKey): propName is ForwardedProps
 }
 
-export interface StyledOptions<Props> {
+export interface StyledOptions<Props = Record<PropertyKey, any>> {
```

Comment:
well, this generic stays unused so I think it should be more than OK to just drop it

Reply:
Yep. I suggest using this impl cause it's compatible with both v10 and v11-based app code. Removing generic will be a breaking change for someone.

Reply:
Any change is a breaking change for someone 😅 I think in this case it's a pragmatic choice to not treat it as one - especially give that I don't expect `StyledOptions` to be used explicitly in a lot of codebases.

Reply:
Maybe it's better to use generic argument if passed?

```ts
export interface StyledOptions<Props = null> {
  label?: string
  shouldForwardProp?(propName: Props extends null ? PropertyKey : keyof Props): boolean
  target?: string
}
```

Reply:
Quite honestly, I'm not sure if it's worth it. This type should not be used that much publicly and we'd have to check if this would be compatible when passed to `styled` calls. It would also be good to dig up the PRs by @JakeGinnivan which were related to this and make sure that this would not break because of the reasons mentioned there. Ideally, we would pass `Props` to `StyledOptions` from within `CreateStyledComponent` types.

If you are willing to do the work - then sure, ye, we could try to do that.

Reply:
I have a case where it makes sense. We'd like to reuse some options and corresponding types for our UI kit primitives and rich components.

https://github.com/qiwi/pijma/blob/d884ba8b954b59f190623caf1623c9489c11b550/packages/core/src/primitive/Card.ts#L23
```ts
export interface CardProps extends BoxProps {
  bg?: string
  s?: string
  b?: string
  bt?: string
  br?: string
  bb?: string
  bl?: string
  r?: Value
  rtr?: Value
  rtl?: Value
  rbr?: Value
  rbl?: Value
}

export const CardNonProps = ['bg', 'b', 'bt', 'br', 'bb', 'bl', 'r', 'rtr', 'rtl', 'btr', 'btl', 's'].concat(BoxNonProps)

export const CardOptions: StyledOptions<CardProps> = {
  // expect any prop key strictly belong to CardProps only
  shouldForwardProp: (prop) => !CardNonProps.includes(prop),
}
```

Reply:
The question is - can the resulting type be used when using `styled`?

```ts
type Props = { foo: boolean }

declare const options: StyledOptions<Props>

const test1 = styled.div<Props>({
  color: 'red',
}, options)

const test2 = styled.div({
  color: 'red',
}, options)
```

Reply:
Everything should be ok. There's a minor type issue, which is not related to this PR.
```
TS2769: No overload matches this call.   Overload 1 of 3, '(...styles: Interpolation<{ theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement> & { ...; }>[]): StyledComponent<...>', gave the following error.     Argument of type 'StyledOptions<Props>' is not assignable to parameter of type 'Interpolation<{ theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<HTMLDivElement> & { ...; }>'.       Type 'StyledOptions<Props>' is not assignable to type 'CSSObject'.   Overload 2 of 3, '(template: TemplateStringsArray, ...styles: Interpolation<{ theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<...> & { ...; }>[]): StyledComponent<...>', gave the following error.     Argument of type '{ color: string; }' is not assignable to parameter of type 'TemplateStringsArray'.       Object literal may only specify known properties, and 'color' does not exist in type 'TemplateStringsArray'.   Overload 3 of 3, '(template: TemplateStringsArray, ...styles: Interpolation<{ theme?: Theme | undefined; as?: ElementType<any> | undefined; } & ClassAttributes<HTMLDivElement> & HTMLAttributes<...> & { ...; }>[]): StyledComponent<...>', gave the following error.     Argument of type '{ color: string; }' is not assignable to parameter of type 'TemplateStringsArray'.       Object literal may only specify known properties, and 'color' does not exist in type 'TemplateStringsArray'.
```

Reply:
Well, if this is supposed to land then I strongly believe that those 2 cases should not result in any type errors - even if the problem is not related to this PR it IMHO would be preferable to fix it here or prepare a separate fix, land it and then rebase this PR. Without those working it's hard for me to justify merging this PR with that generic being used.

Reply:
Give me a bit more time. I'll fix the issue above.

Reply:
No problem - there is no rush. I appreciate your efforts here :)

Reply:
@Andarist, 

Not sure if I found the best solution, but it seems to work.

Reply:
@Andarist,
Could you take a look?